store/src/java/com/zimbra/cs/mime/MimeUtil.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2017 Synacor, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+package com.zimbra.cs.mime;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+
+import javax.mail.internet.MimeUtility;
+
+import com.sun.mail.util.BEncoderStream;
+import com.sun.mail.util.PropUtil;
+import com.sun.mail.util.QEncoderStream;
+
+public class MimeUtil {
+    /** (non-Javadoc)
+     * Encode <tt>string</tt> no matter what characters are contained in it.
+     * @param string text to be encoded
+     * @param charset if it's set to <tt>null</tt>, UTF-8 is chosen as default
+     * @param encoding if it's set to <tt>null</tt>, base64 encoding is chosen as default
+     * @param encodingWord available when encoding is "quoted-printable". If it's set to <tt>false</tt>,
+     * only "=_?" characters are encoded, otherwise "=_?\"#$%&'(),.:;<>@[\\]^`{|}~" will be encoded.
+     * @return encoded text
+     * @throws UnsupportedEncodingException
+     *
+     * @see  javax.mail.internet.MimeUtility.encodeWord(String string, String charset, String encoding, boolean encodingWord)
+     */
+    public static String encodeWord(String string, String charset,
+            String encoding, boolean encodingWord) throws UnsupportedEncodingException {
+        if (!string.contains("\r") && !string.contains("\n")){
+            String result = string;
+            try {
+                result = MimeUtility.encodeText(string);
+            } catch (UnsupportedEncodingException e) {
+                // ignore the exception and return the source string as is.
```

Comment:
Do we need to log a message here? At lease at debug level.

Reply:
I don't think any log message is necessary; if the encoding is failed, I will see the un-encoded string in the resulting message.